platforms/jvm/language-java/src/main/java/org/gradle/api/plugins/jvm/internal/JvmFeatureInternal.java

```diff
@@ -91,15 +93,15 @@ public interface JvmFeatureInternal extends Named {
      * @return null if {@link #withJavadocJar()} has not been called.
      */
     @Nullable
-    Configuration getJavadocElementsConfiguration();
+    NamedDomainObjectProvider<ConsumableConfiguration> getJavadocElementsConfiguration();
 
     /**
      * Gets the consumable configuration created by {@link #withSourcesJar()}.
      *
      * @return null if {@link #withSourcesJar()} has not been called.
      */
     @Nullable
-    Configuration getSourcesElementsConfiguration();
+    NamedDomainObjectProvider<ConsumableConfiguration> getSourcesElementsConfiguration();
```

Comment:
A nullable provider is uncommon, it be non nullable but without a value instead.